### PR TITLE
document values clause and add it to SELECT

### LINKED ISSUE #763: `values` is enabled in materialized view
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/8751

This pr is part of https://github.com/risingwavelabs/risingwave/issues/7857

### Which part(s) of the docs might be affected or should be updated? And how?

Release notes:
`values` are available in materialized views.

Update https://www.risingwave.dev/docs/current/sql-create-mv/

### Reference

_No response_

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Document the VALUES clause and mention it can be used in SELECT

- **Preview**: 
[ Paste the preview link to the edited page here. ]

- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves #763

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/sql/commands/sql-select.md [hunk 1]
```diff
@@ -31,7 +35,9 @@ Where `from_item` can be:
 
 |Parameter or clause        | Description           |
 |---------------------------|-----------------------|
+|**WITH** clause           | Provides a way to write supplemental statements for a larger query. For more information, see [`WITH` clause](/sql/query-syntax/query-syntax-with-clause.md). |
 |*expression*               |A column or an expression.|
+|**VALUES** clause          | This clause generate one or more rows of data as a table expression. For details, see [VALUES clause](../query-syntax/query-syntax-values-clause.md).|
```
Comment (emile-00) [inline docs/sql/commands/sql-select.md]: generate -> generates

**Review by emile-00 — APPROVED** [review]

File: docs/sql/query-syntax/query-syntax-values-clause.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+id: query-syntax-values-clause
+slug: /query-syntax-values-clause
+title: VALUES clause
+---
+
+In RisingWave, the `VALUES` clause is used to generate one or more rows of data as a table expression. It is commonly used in SQL queries to create temporary tables or to insert data into a table or materialized view.
```
Comment (y-wei) [inline docs/sql/query-syntax/query-syntax-values-clause.md]: AFAIK current risingwave does not  support insert into materialized views?
```SQL
dev=> insert into mv values (2);
ERROR:  QueryError: Invalid input syntax: cannot change materialized view "mv"
```

File: docs/sql/query-syntax/query-syntax-values-clause.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+Here, each set of expressions enclosed in parentheses represents a row of data. The number of expressions must match the number of columns in the table or view being created.
+
+To use the `VALUES` clause in creating a materialized view, you can specify it in the `SELECT` statement used to define the view. For example:
+
+```sql
+CREATE MATERIALIZED VIEW my_view AS
+  SELECT column1, column2
+  FROM (VALUES (1, 'John'), (2, 'Jane'), (3, 'Bob')) AS my_table(column1, column2);
```
Comment (y-wei) [inline docs/sql/query-syntax/query-syntax-values-clause.md]: This is a viable approach, yet there is a simpler one:
`create materialized view mvv (a, b) as values (1, 'John'), (2, 'Jane');`

File: docs/sql/query-syntax/query-syntax-values-clause.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+In this example, the `VALUES` clause is used to generate a temporary table with three rows and two columns. The resulting table is then used in the `SELECT` statement to create a materialized view called `my_view` with columns `column1` and `column2`.
+
+The `VALUES` clause can also be used in more complex queries, such as subqueries or joins, to generate temporary tables or insert data into existing tables. For example:
+
+```sql
+CREATE MATERIALIZED VIEW mv AS
+WITH dict(abbr, full) AS (VALUES ('cn', 'China'), ('us', 'United States')) 
+SELECT * FROM t JOIN dict ON t.c = dict.real;
+```
```
Comment (y-wei) [inline docs/sql/query-syntax/query-syntax-values-clause.md]: Would it be better to write a command using the above materialized view?

**Comment by hengm3467** [comment]
@Eurekaaw I implemented your comments. Please take a look again. Thanks.

**Review by y-wei — APPROVED** [review]
LGTM

**Pull request merged**